bloom.js

```diff
@@ -37,43 +37,30 @@ JSBloom.filter = function(items, target_prob) {
             };
 
             return (hash >>> 0) % BUFFER_LEN;
-        },
-        getIndices: function(str) {
-
-            var hashes = [];
-
-            hashes.push(this.djb2(str));
-            hashes.push(this.sdbm(str));
-
-            for (var round = 2; round <= HASH_ROUNDS; round++) {
-                var new_hash = (hashes[0] + (round * hashes[1]) + (round^2)) % BUFFER_LEN;
-
-                hashes.push(new_hash);
-            };
-
-            return hashes;
-
         }
     }
 
     addEntry = function(str) {
 
-            var indices = hashes.getIndices(str);
-
-            for (var i = indices.length - 1; i >= 0; i--) {
+        var h1 = hashes.djb2(str)
+        var h2 = hashes.sdbm(str)
+        for (var round = 0; round <= HASH_ROUNDS; round++) {
+            var new_hash = round == 0 ? h1
```

Comment:
Yes, definitely. It's unfortunate that the fastest code is often not the prettiest code.